app/src/main/java/org/p2p/wallet/history/repository/TransactionConverter.kt

```diff
@@ -32,34 +35,38 @@ object TransactionConverter {
         HistoryTransaction.Swap(
             signature = response.signature,
             sourceAddress = sourcePublicKey,
-            destinationAddress = response.destination,
+            destinationAddress = response.destination.orEmpty(),
             fee = response.fee.toBigInteger(),
             blockNumber = response.slot,
             date = ZonedDateTime.ofInstant(
                 Instant.ofEpochMilli(response.getBlockTimeInMillis()),
                 ZoneId.systemDefault()
             ),
             amountA = response.amountA
-                .toBigInteger()
-                .fromLamports(sourceData.decimals)
-                .scaleLong(),
+                ?.toBigInteger()
```

Comment:
```
?.toBigInteger()
                ?.fromLamports(destinationData.decimals)
                ?.scaleLong()
                ?: BigDecimal.ZERO
```

Reply:
I suggest to move repeating code into function to avoid boilepart

Reply:
will do, thanks